### DISCUSSION TITLE
Unique Id is not getting generated via Open Telemetry. Getting repeated trace Id after 10 iterations

### DISCUSSION THREAD

**Discussion by keerthanacv** [open-telemetry/opentelemetry-java-instrumentation#11951] [Q&A] [answered]
@Bean
public static OpenTelemetry openTelemetry() {
    Resource resource = Resource.getDefault();
    SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
                 .addSpanProcessor(SimpleSpanProcessor.create(LoggingSpanExporter.create()))
            .setResource(resource)
            .build();

    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
            .registerMetricReader(PeriodicMetricReader.builder(LoggingMetricExporter.create()).build())
            .setResource(resource)
            .build();

    SdkLoggerProvider sdkLoggerProvider = SdkLoggerProvider.builder()
            .addLogRecordProcessor(BatchLogRecordProcessor.builder(SystemOutLogRecordExporter.create()).build())
            .setResource(resource)
            .build();

    OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
            .setTracerProvider(sdkTracerProvider)
            .setMeterProvider(sdkMeterProvider)
            .setLoggerProvider(sdkLoggerProvider)
            .setPropagators(ContextPropagators.create(TextMapPropagator.composite(W3CTraceContextPropagator.getInstance(), W3CBaggagePropagator.getInstance())))
            .buildAndRegisterGlobal();

    return openTelemetry;
}
In every controller and service, I added this

private Tracer tracer;

private OpenTelemetry openTelemetry;

AccountInfoController(OpenTelemetry openTelemetry) {

    tracer = openTelemetry.getTracer(AccountInfoController.class.getName(), "0.1.0");
}
and created span in Controller as below

Span span = tracer.spanBuilder(spanName).startSpan();
(where span name is the controller name) and in the service class I created span

Context.current().with(parentSpan).makeCurrent();
Span serviceSpan = tracer.spanBuilder(spanName)
        .setParent(Context.current().with(Span.current()))
        .startSpan();
where parent span is controller span

However, I'm receiving the trace Id repeated after 10 iterations

How to generate unique trace IDs?

**Comment by laurit** [comment, marked as answer]
This is a coding error on your part. The scope returned from `Context.current().with(parentSpan).makeCurrent();` must be closed, it is best used with try with resource block.

**Comment by keerthanacv** [comment]
By closing the Context.current().with(parentSpan).makeCurrent(); fixed the issue
